### DISCUSSION TITLE
Is there a way to manage container memory limit?

### DISCUSSION THREAD

**Discussion by mattupstate** [rancher-sandbox/rancher-desktop#3301] [Q&A] [answered]
I'm using Rancher Desktop for local development on macOS Monterey. I'm trying to build an application image (complex NextJS application) that requires more memory than what Rancher Desktop affords to containers by default (it seems to be about 3Gb). I thought that the `-m/--memory` flag for `docker build` would help, but the limit does not change according to what I see when I run `docker stats` as the image builds. Is there a way to increase this limit at all?

**Comment by jandubois** [comment, marked as answer]
Are you running on Intel or ARM? There is a limitation to 3GB on ARM when running on macOS versions older than 12.4 due to a macOS kernel bug that has been fixed by Apple in 12.4.

If you are not on ARM/M1, or on a newer version of macOS, then your issue must be something else.

**Reply by mattupstate** [reply]
Ah yes, I forgot to mention that detail. I am indeed on ARM 😢 Thanks for the prompt reply!

**Reply by jandubois** [reply]
This should only affect you if you are still running an old version of Monterey. I think the latest is 12.6, so if you install the latest updates, you should be fine.

**Reply by mattupstate** [reply]
I was on 12.3. Just updated to 12.6 and now the limit is about 8Gb, which coincidentally is the value I set for the Rancher Desktop VM memory.

**Reply by jandubois** [reply]
> now the limit is about 8Gb, which coincidentally is the value I set for the Rancher Desktop VM memory.

It is not coincidental; this is directly related, as the container runs inside the VM. If you give the VM more memory, it should be able to run correspondingly larger containers...
